ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -0,0 +1,311 @@
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default GeoHeatmapSeries;
+
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+/**
+ * A `geoheatmap` series. If the [type](#series.map.type) option is not
+ * specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.geoheatmap
+ * @excluding allAreas, dataParser, dataURL, dragDrop, findNearestPointBy,
+ *            joinBy, marker, mapData, negativeColor, onPoint, shadow
+ * @product   highmaps
+ * @apioption series.geoheatmap
+ */
+
+/**
+ * An array of data points for the series. For the `geoheatmap` series
+ * type, points can be given in the following ways:
+ *
+ * 1.  An array of arrays with 3 or 2 values. In this case, the values
+ * correspond to `lon,lat,value`. The `value` refers to the color on the `colorAxis`.
+ *
+ *  ```js
+ *     data: [
+ *         [51.50, -0.12, 7],
+ *         [54.59, -5.93, 4],
+ *         [55.8, -4.25, 3]
+ *     ]
+ *  ```
+ *
+ * 2.  An array of objects with named values. The following snippet shows only a
+ * few settings, see the complete options set below. If the total number of data
+ * points exceeds the series' [turboThreshold](#series.heatmap.turboThreshold),
+ * this option is not available.
+ *
+ *  ```js
+ *     data: [{
+ *         lat: 51.50,
+ *         lon: -0.12,
+ *         value: 7,
+ *         name: "London"
+ *     }, {
+ *         lat: 54.59,
+ *         lon: -5.93,
+ *         value: 4,
+ *         name: "Belfast"
+ *     }]
+ *  ```
+ *
+ * @sample maps/demo/geoheatmap/
+ *         GeoHeatmap chart
+ *
+ * @type      {Array<Array<number>|*>}
+ * @extends   series.map.data
+ * @product   highmaps
+ * @apioption series.geoheatmap.data
+ */
+
+/**
+ * Individual color for the point. By default the color is either used
+ * to denote the value, or pulled from the global `colors` array.
+ *
+ * @type      {Highcharts.ColorString|Highcharts.GradientColorObject|Highcharts.PatternObject}
+ * @product   highmaps
+ * @apioption series.geoheatmap.data.color
+ */
+
+/**
+ * The value of the point, resulting in a color controled by options
+ * as set in the [colorAxis](#colorAxis) configuration.
+ *
+ * @type      {number|null}
+ * @product   highmaps
+ * @apioption series.geoheatmap.data.value
+ */
+
+/**
+ * The border width of each geoheatmap tile.
+ *
+ * In styled mode, the border stroke width is given in the
+ * `.highcharts-point` class.
+ *
+ * @type      {number|null}
+ * @default   1
+ * @product   highmaps
+ * @apioption series.geoheatmap.borderWidth
+ */
+
+/**
+ * The column size - how many longitude units each column in the
+ * geoheatmap should span.
+ *
+ * @type      {number|null}
+ * @default   1
+ * @product   highmaps
+ * @apioption series.geoheatmap.colsize
+ */
+
+/**
+ * The rowsize size - how many latitude units each row in the
+ * geoheatmap should span.
+ *
+ * @type      {number|null}
+ * @default   1
+ * @product   highmaps
+ * @apioption series.geoheatmap.rowsize
+ */
```

Comment:
These options are defined under `plotOptions.geoheatmap.xxx` above, and we shouldn't have to declare them again here. Inheritance should be taken care of by `@extends` on line 211.